File: antson.java/src/test/java/io/odysz/semantic/jprotocol/test/AnsonBody.java
package io.odysz.semantic.jprotocol.test;

import io.odysz.anson.Anson;
import io.odysz.anson.AnsonField;

public abstract class AnsonBody extends Anson {
	public static String[] jcondt(String logic, String field, String v, String tabl) {
		return new String[] {logic, field, v, tabl};
	}

	@AnsonField(ref=AnsonField.enclosing)
	public AnsonMsg<? extends AnsonBody> parent;

	protected String uri;
	public String uri() { return uri; }

	/** Action: login | C | R | U | D | any serv extension */
	String a;
	/** @return Action: login | C | R | U | D | any serv extension */
	public String a() { return a; }

	public AnsonBody a(String act) {
		this.a = act;
		return this;
	}

	protected AnsonBody(AnsonMsg<? extends AnsonBody> parent, String uri) {
		this.parent = parent;
		this.uri = uri;
	}
}

File: antson.csharp/TestAnson.net/semantics/io.odysz/jprotocol/AnsonBody.cs
using io.odysz.anson;

namespace io.odysz.semantic.jprotocol
{
	public class AnsonBody : Anson
	{
		/// <summary>
		/// @AnsonField(ref= AnsonField.enclosing)
		/// </summary>
		[AnsonField(refer=AnsonField.enclosing)]
		public AnsonMsg parent { get; internal set; }

		public AnsonBody Parent(AnsonMsg p)
        {
			parent = p;
			return this;
        }

		public string conn { get; protected set; }

		/// <summary>
		/// Action: login | C | R | U | D | any serv extension
		/// </summary>
		protected string a { get; set; }

        /// <summary>
        /// @return Action: login | C | R | U | D | any serv extension
        /// </summary>
        public string A() { return a; }

        public AnsonBody A(string act)
        {
            a = act;
            return this;
        }

        public AnsonBody(AnsonMsg parent, string conn)
		{
			this.parent = parent;
			this.conn = conn;
		}
	}
}
